examples/auctions/blind_auction.vy

```diff
@@ -43,127 +39,90 @@ def __init__(_beneficiary: address, _biddingTime: timedelta, _revealTime: timede
 
 # Place a blinded bid with:
 #
-# _blindedBid = keccak256(concat(
-#       convert(value, bytes32),
-#       convert(fake, bytes32),
-#       secret)
-# )
-#
-# The sent ether is only refunded if the bid is correctly revealed in the
-# revealing phase. The bid is valid if the ether sent together with the bid is
-# at least "value" and "fake" is not true. Setting "fake" to true and sending
-# not the exact amount are ways to hide the real bid but still make the
-# required deposit. The same address can place multiple bids.
+# @param _blindedBib a blinded bid equal to:
+#        keccak256(concat(convert(bid_amount, bytes32),bidder,secret))
 @public
 @payable
-def bid(_blindedBid: bytes32):
+def submitBlindedBid(_blindedBid: bytes32):
     # Check if bidding period is still open
     assert block.timestamp < self.biddingEnd
 
-    # Check that payer hasn't already placed maximum number of bids
-    numBids: int128 = self.bidCounts[msg.sender]
-    assert numBids < MAX_BIDS
+    # Check that this bid was not accepted already
+    assert self.bids[_blandedBid] == 0
+
+    # Check that some funds are committed
+    assert msg.value > 0
+
+    # Register the bid
+    self.bids[_blandedBid] = msg.value
+
+
+# Reveal a bid during the bid reveal period
+#
+# @param  _secret    a secret value used as part of the blinded bid
+# @param  _bidAmount the effective bid amount
+# @return True if bid is not the highest bid, False otherwise
+@public
+def revealBid(_secret: bytes32, _bidAmount: wei_value) -> bool:
+    # Check that bidding period is over
+    assert block.timestamp > self.biddingEnd
 
-    # Add bid to mapping of all bids
-    self.bids[msg.sender][numBids] = Bid({
-        blindedBid: _blindedBid,
-        deposit: msg.value
-        })
-    self.bidCounts[msg.sender] += 1
+    # Decode blinded bid
+    blindedBid: bytes32 = keccak256(concat(
+        convert(_bidAmount, bytes32),
+        convert(msg.sender, bytes32),
+        _secret
+    ))
 
+    # Handle a valid new highest bid
+    if block.timestamp < self.revealEnd
+        if _bidAmount > self.highestBid
+            if self.bids[blindedBid] >= _bidAmount
+                # Refund prior locked-up funds
+                # Security notice: overflow on += is not a concern because the
+                # amount of Ether in exististance fits in 256-bit number
```

Comment:
Security notice not necessary, Vyper has overflow protection by default

Reply:
Fixed in 2ce9488961378c2439680fd62353e60e9f0c49c5